### ISSUE TITLE
Fix test_platform linker issues on Ubuntu

### ISSUE THREAD

**Issue by devhawk** [Microsoft/xlang#46]
The [fix_ubuntu](https://github.com/Microsoft/xlang/tree/fix_ubuntu) branch now compiles the PAL clean on Ubuntu (we were missing an include file). However, test_platform fails to link with undefined references:

```
platform/libpal.so: undefined reference to `xlang::impl::convert_string(char const*, unsigned int, char16_t*, unsigned int)'
platform/libpal.so: undefined reference to `xlang::impl::get_converted_length(char16_t const*, unsigned int)'
platform/libpal.so: undefined reference to `xlang::impl::get_converted_length(char const*, unsigned int)'
platform/libpal.so: undefined reference to `xlang::impl::convert_string(char16_t const*, unsigned int, char*, unsigned int)'
```

**Comment by devhawk** [comment]
Spoke to Ryan - "The link error is straightforward: we simply need a non-windows implementation of utf-8 to/from utf-16"

**Comment by devhawk** [comment]
fixed by #48 (in PR)
